types/concurrency-limiter/limiter.d.ts

```diff
@@ -0,0 +1,46 @@
+/**
+  The concurrency limiter.
+
+  It is instantiated providing the maximum number of async tasks that are
+  allowed to run concurrently.
+*/
+declare class Limiter {
+    constructor(limit: number);
+
+    private _limit: number;
+    private _length: number;
+    private _queue: Array<() => void>;
+
+    /**
+	Request a slot for concurrent execution. The returned promise is only
+	resolved when a slot becomes available. The promise is never rejected.
+	This method is usually called before starting an async operation, like:
+
+	await limiter.enter();
+	await wsConnect();
+	...
+	wsClose();
+	limiter.exit();
+
+	@returns A promise resolved when a slot is available.
+	*/
+    enter(): Promise<void>;
+
+    /**
+	Release a slot for concurrent execution. This method is usually called
+	after an async operation completes.
+	*/
+    exit(): void;
+
+    /**
+	Execute the provided async function when a concurrent slot is available,
+	and release the slot when the function completes, returning its result.
+	The limiter is exited even in the case the function raises an exception.
+
```

Comment:
```suggestion

	@async
```

Reply:
Because of my poor knowledge of JSDoc, I don't really know what effect this '@async' does. Could you please tell me why this '@async' is needed?

Reply:
there is no effect, it's just TS does not allow `async` in types declarations, so `@async` (as per JSDoc) informs method has been declared using `async`.
Just nice touch for users

Reply:
I see. It seems that it's truly a good way to inform. Thank you!